src/db/tables/subscribers.js

```diff
@@ -65,6 +65,7 @@ async function getSubscriberById (id) {
 
 /**
  * @param {string} uid
+ * @returns {Promise<undefined | import("knex/types/tables").SubscriberRow & { email_addresses: Array<{ id: import("knex/types/tables").EmailAddressRow["id"]; email: import("knex/types/tables").EmailAddressRow["email"]; }> }>}
```

Comment:
I started out by adding the `@returns` annotation here (in one commit, and on line 82 in another), and then following up the TypeScript warnings. The other changes should only be type-level changes (i.e. they don't affect how our code runs), except where it explicitly handles the case where this returns `undefined`, logging an error. Hopefully, that will make it easier to debug issues down the road.

(Also: I wish this had been covered by TypeScript in from the start, but alas...)